File: src/main/java/org/itmo/isLab1/locations/LocationController.java
package org.itmo.isLab1.locations;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;
import org.itmo.isLab1.common.framework.CrudController;
import org.itmo.isLab1.locations.dto.*;

@RestController
@RequestMapping(value = "/api/locations", produces = MediaType.APPLICATION_JSON_VALUE)
public class LocationController
    extends CrudController<
        Location,
        LocationDto,
        LocationCreateDto,
        LocationUpdateDto,
        LocationService> {

    public LocationController(
        LocationService service
    ) {
        super(service);
    }
}

File: frontend/src/controllers/LocationController.ts
import { LocationDto } from '@/interfaces/dto/locations/LocationDto';
import { LocationCreateDto } from '@/interfaces/dto/locations/LocationCreateDto';
import { LocationUpdateDto } from '@/interfaces/dto/locations/LocationUpdateDto';
import CrudController from '@/interfaces/crud/CrudController';


export default new CrudController<LocationDto, LocationCreateDto, LocationUpdateDto>("locations")
